### ISSUE TITLE
Set as default browser 

### ISSUE THREAD

**Issue by 118-61-21-230** [mises-id/mises-browser-core#212]
Even IF you set the browser as default, still that  option is visible. 

![Screenshot_2024-01-21-17-57-05-62_d4b821f22c36211daa9077338f9854d0](https://github.com/mises-id/mises-browser-core/assets/140193418/925ef030-18fa-4d44-843f-c04f9b2aad02)

**Comment by Haoxiqiang** [comment]
This is not entirely Mises' problem.
In the Android platform, detecting who is the default browser is not accurate enough.

`resolveInfo.activityInfo` may return `android` or `site.mises.browser`.

should consider whether to relax some conditions.

```kotlin
// first time use strict
if (strict) {
    context.packageName == resolveInfo.activityInfo.packageName
} else {
    context.packageName == resolveInfo.activityInfo.packageName ||
        context.packageName == "android"
}
```

**Comment by freshlife001** [comment]
fixed in #221
